### PR TITLE
Move/remove old information from "What's new" page

### PR DESCRIPTION
**Pull request by SimonHeybrock**
- Move copy section to reference docs (reformulated slightly, to not refer to "old", "new", or "change" any more)
- Remove subsections about additions that did not see changes after 0.6

### CODE REVIEW

File: docs/reference/ownership-mechanism-and-readonly-flags.ipynb [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "be0a620c-db87-4c06-967a-ff1ec260c567",
+   "metadata": {},
+   "source": [
+    "## Data arrays\n",
+    "\n",
+    "As a result of the sharing mechanism, extra care must now be taken in some cases, just like when working with any other Python library.\n",
```
Comment (jl-wynen) [inline docs/reference/ownership-mechanism-and-readonly-flags.ipynb:80]:
```suggestion
    "As a result of the sharing mechanism, extra care must be taken in some cases, just like when working with any other Python library.\n",
```

File: docs/reference/ownership-mechanism-and-readonly-flags.ipynb [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "98faae5d-4251-409f-87f7-9c5675ccdeb6",
+   "metadata": {},
+   "source": [
+    "Apart from the standard and pythonic behavior, one advantage of this is that creating data arrays from variables is typically cheap, without inflicting copies of potentially large objects.\n",
+    "\n",
+    "A related concept are read-only flags.\n",
```
Comment (jl-wynen) [inline docs/reference/ownership-mechanism-and-readonly-flags.ipynb:144]: Please make this a sub(sub)section to make it easier to find if someone has questions.

**Review by jl-wynen — APPROVED** [review]
Just some minor language / layout comments. Looks good otherwise.

File: docs/reference/ownership-mechanism-and-readonly-flags.ipynb (hunk 2 above)
Comment (SimonHeybrock) [reply docs/reference/ownership-mechanism-and-readonly-flags.ipynb]: Extracted the whole paragraph into new section.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**